src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -276,7 +276,10 @@ ObjectMonitor::ObjectMonitor(oop object) :
 { }
 
 ObjectMonitor::~ObjectMonitor() {
-  _object.release(_oop_storage);
+  if (!_object.is_null()) {
+    // Release object's oop storage if it hasn't already been done.
+    release_object();
+  }
```

Comment:
In what case, would the object's oop storage not already been released?  Is the destructor called here because of GC, where the object has been collected?